src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -47,61 +68,185 @@ public class EditCommand extends Command {
+                    + "Parameters: INDEX (must be a positive integer) "
+                    + "[" + PREFIX_NAME + "NAME] "
+                    + "[" + PREFIX_PHONE + "PHONE] "
+                    + "[" + PREFIX_EMAIL + "EMAIL] "
+                    + "[" + PREFIX_ADDRESS + "ADDRESS] "
+                    + "[" + PREFIX_QUALIFICATION + "QUALIFICATION] "
+                    + "[" + PREFIX_INSTITUTION + "INSTITUTION] "
+                    + "[" + PREFIX_TAG + "TAG]...\n"
+                    + "Example: " + COMMAND_WORD + " 1 "
+                    + PREFIX_PHONE + "91234567 "
+                    + PREFIX_EMAIL + "johndoe@example.com "
+                    + PREFIX_QUALIFICATION + "MSc";
+
+        case TUITIONCLASS_LIST:
+            return COMMAND_WORD + ": Edits the details of the class identified "
+                    + "by the index number used in the displayed class list. "
+                    + "Existing values will be overwritten by the input values.\n"
+                    + "Parameters: INDEX (must be a positive integer) "
+                    + "[" + PREFIX_NAME + "NAME] "
+                    + "[" + PREFIX_SUBJECT + "SUBJECT] "
+                    + "[" + PREFIX_LEVEL + "LEVEL] "
+                    + "[" + PREFIX_DAY + "DAY] "
+                    + "[" + PREFIX_TIME + "TIME] "
+                    + "[" + PREFIX_TAG + "TAG]...\n"
+                    + "Example: " + COMMAND_WORD + " 1 "
+                    + PREFIX_SUBJECT + "Mathematics "
+                    + PREFIX_DAY + "Monday "
+                    + PREFIX_TIME + "12:00-14:00";
+
+        default:
+            return MESSAGE_USAGE;
+        }
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        List<Person> lastShownList = model.getFilteredPersonList();
+
+        FilteredList<?> lastShownList = model.getCurrentList();
 
         if (index.getZeroBased() >= lastShownList.size()) {
             throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
         }
 
-        Person personToEdit = lastShownList.get(index.getZeroBased());
-        Person editedPerson = createEditedPerson(personToEdit, editPersonDescriptor);
+        if (model.getCurrentListType() == Model.ListType.TUITIONCLASS_LIST) {
+            TuitionClass classToEdit = (TuitionClass) lastShownList.get(index.getZeroBased());
+            TuitionClass editedClass = createEditedClass(classToEdit, editDescriptor);
+
+            if (!classToEdit.isSameTuitionClass(editedClass) && model.hasTuitionClass(editedClass)) {
+                throw new CommandException(MESSAGE_DUPLICATE_CLASS);
+            }
+
+            model.setTuitionClass(classToEdit, editedClass);
+            model.updateFilteredTuitionClassList(PREDICATE_SHOW_ALL_TUITIONCLASS);
+            return new CommandResult(String.format(MESSAGE_EDIT_CLASS_SUCCESS, editedClass));
+        }
+
```

Comment:
Maybe can assert(lastShownList instanceof FilteredList<Student> || lastShownList instanceof FilteredList<Tutor>) or assert (model.getCurrentListType() == Model.ListType.STUDENT_LIST || model.getCurrentListType() == Model.ListType.TUTOR_LIST) here so that it is safer if we were to modify list type in the future(currently I think we are planning to make Person class abstract so PERSON_LIST  would be removed)